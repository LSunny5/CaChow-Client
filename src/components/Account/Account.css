/*entire account page*/
.AccountPage {
    min-height: 91vh;
    padding-bottom: 1rem;
}

/*welcome message*/
.formTitle {
    width: 90%;
    font-size: 2rem;
    margin: 0 auto;
    padding: 1rem;
    font-weight: bolder;
    font-family: 'Cinzel', serif;
}

/*add restaurant button*/
.addRestButton {
    display: block;
    width: 60%;
    margin: 0 auto 1.5rem;
    background-color: lightgreen;
    text-decoration: none;
    border: 1px solid black;
    padding: 10px;
    border-radius: 10px;
}

/*container for restaurant and buttons associated*/
.restaurantBox {
    padding: 3%;
    margin: 0 auto;
    width: 90%;
    border-bottom: 1px dotted rgb(2, 34, 2);
    font-family: 'Delius', cursive;
}

/*div for restaurant title*/
.RestaurantName {
    margin: 1rem auto;
}

/*restaurant title text*/
.rView {
    font-size: 2em;
    font-weight: bold;
    font-family: 'Cinzel', serif;
    line-height: 1rem;
}

/*message for when no restaurants added*/
.noRestaurants {
    margin: 20px auto;
    border: 1px dotted green;
    padding: 2rem;
    width: 70%;
    font-size: 1.2rem;
    font-family: 'Delius', cursive;
    background: rgb(255, 255, 248);
}

/*div for button boxes*/
.buttonsboxes {
    margin: 1.3rem auto;
    display: flex;
    flex-direction: row;
    flex-wrap: wrap;
}

/*button look on account page*/
.ownerButton {
    display: block;
    text-decoration: none;
    border: 1px solid black;
    color: black;
    padding: 1rem;
    margin: 0.5rem auto;
    border-radius: 10px;
    cursor: pointer;
    width: 40%;
    background: white;
    font-weight: bolder;
    transition: 1s;
}

/*button on hover*/
.ownerButton:hover {
    background-color: black;
    color: white;
}

/*User remove button*/
.removeButton {
    color: red;
    font-weight: bolder;
    background: lightpink; 
    border-radius: 3px;
    padding: 1.5rem auto;
    width: 50%;
    font-family: 'Delius', cursive;
    border: none;
    cursor: pointer;
}

/*hover of remove button*/
.removeButton:hover {
    background-color: red;
    color: lightcoral;
}

/*make text red and bold*/
.redandbold {
    color: red;
    font-weight: bold;
}

/*make text green*/
.green {
    color: green;
}

/********************************************************************/
/*For Ipads and larger screens */
@media only screen and (min-width: 600px) {
    /*entire account page*/
    .AccountPage {
        padding-top: 1rem;
    }
    
    /*welcome message*/
    .formTitle {
        font-size: 3rem;
    }

    /*add restaurant button*/
    .addRestButton {
        width: 20%;
        margin: 0 auto;
    }

    /*div for restaurant title*/
    .RestaurantName {
        margin: 2rem auto;
    }

    /*message for when no restaurants added*/
    .noRestaurants {
        margin: 3rem auto 1rem;
    }

    /*div for button boxes*/
    .buttonsboxes {
        margin: 1rem auto;
    }

    /*button look on account page*/
    .ownerButton {
        padding: 1rem;
        margin: 0.5rem auto;
        border-radius: 10px;
        width: 22%;
    }

    /*User remove button*/
    .removeButton {
        width: 25%;
    }
}

/********************************************************************/
/*For Laptops and larger screens */
@media only screen and (min-width: 1024px) {
    /*welcome message*/
    .formTitle {
        font-size: 3rem;
    }

    /*add restaurant button*/
    .addRestButton {
        width: 15%;
    }

    /*container for restaurant and buttons associated*/
    .restaurantBox {
        padding: 1rem;
    }

    /*div for button boxes*/
    .buttonsboxes {
        margin: 1rem auto;
    }

    /*button look on account page*/
    .ownerButton {
        padding: 1rem;
        margin: 0 auto;
        width: 18%;
    }

    /*User remove button*/
    .removeButton {
        width: 15%;
    }
}