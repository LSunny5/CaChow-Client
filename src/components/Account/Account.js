import React, { Component } from 'react';
import CachowContext from '../../CachowContext';
import { NavLink } from 'react-router-dom';
import './Account.css';
import { getRest } from '../../CachowHelpers';
import PropTypes from 'prop-types';

export default class Account extends Component {
    static contextType = CachowContext;

    render() {
        const { restaurants } = this.context;
        let uname = localStorage.getItem('name');
        let listRest = getRest(restaurants, uname) || {};

        return (
            <section className='AccountPage'>
                <h2 className="formTitle">Welcome {uname}!</h2>
                <div className="otherButtonBox">
                    <NavLink to={`/restaurants/hours/add`}
                        className="addRestButton">
                        + Add Restaurant
                        </NavLink>
                </div>
                {/* create a new box with functions for each of the restaurants for user */}
                {(listRest.length > 0) ? (
                    listRest.map(restaurant =>
                        <div className="restaurantBox" key={restaurant.r_id}>
                            <div className="RestaurantName">
                                <NavLink
                                    to={`/restaurants/${restaurant.r_id}`}
                                    className="rView">
                                    {restaurant.r_name}
                                </NavLink>
                            </div>
                            <div className="buttonsboxes">
                                <NavLink to={`/restaurants/${restaurant.r_id}/edit`}
                                    className="ownerButton">
                                    Edit Restaurant
                                </NavLink>
                                <NavLink to={`/restaurants/${restaurant.r_id}/hours`}
                                    className="ownerButton addRest">
                                    Edit Hours
                                </NavLink>
                                <NavLink to={`/restaurants/${restaurant.r_id}/addItem`} className="ownerButton green">
                                    Add Menu Item
                                </NavLink>
                                <NavLink to={`/restaurants/${restaurant.r_id}/editItems`}
                                    className="ownerButton">
                                    Edit Menu Items
                                </NavLink>
                                <NavLink to={`/restaurants/${restaurant.r_id}/delete`}
                                    className="ownerButton redandbold">
                                    Delete Restaurant
                                </NavLink>
                            </div>
                        </div>
                    )) : (
                        <div className="noRestaurants">
                            You have not added any restaurants at this time...
                        </div>
                    )}
                <br />
                <div className="otherButtonBox">
                    <NavLink to={`/deleteUser`}
                        className="ownerButton removeButton">
                        Remove Account
                    </NavLink>
                </div>
                <br />
                <br />
            </section >
        )
    };
}

Account.propTypes = {
    restaurants: PropTypes.arrayOf(
        PropTypes.shape({
            r_id: PropTypes.number.isRequired,
            r_owner: PropTypes.string.isRequired,
            r_image: PropTypes.string,
            r_type: PropTypes.string,
            r_name: PropTypes.string,
            r_address: PropTypes.string,
            r_city: PropTypes.string,
            r_state: PropTypes.string,
            r_zip: PropTypes.string,
            r_phone: PropTypes.string,
            r_hours: PropTypes.number.isRequired
        })
    ),
}