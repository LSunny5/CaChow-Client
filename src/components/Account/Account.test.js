import React from 'react';
import ReactDOM from 'react-dom';
import Account from './Account';
import toJson from 'enzyme-to-json'
import { shallow } from 'enzyme'
import { BrowserRouter } from 'react-router-dom';

describe(`Account component`, () => {
    const props = {
        r_id: 1,
        r_owner: 'OwnerS',
        r_image: 'NoneS',
        r_type: 'Fast',
        r_name: 'Restaurant Two',
        r_address: 'Address2',
        r_city: 'City2',
        r_state: 'State2',
        r_zip: 'Zip2',
        r_phone: 'Phone2',
        r_hours: 1
    }

    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(
            <BrowserRouter>
                <Account {...props} />
            </BrowserRouter>, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it('renders the Account form', () => {
        const wrapper = shallow(<Account {...props} />)
        expect(toJson(wrapper)).toMatchSnapshot()
    })
});