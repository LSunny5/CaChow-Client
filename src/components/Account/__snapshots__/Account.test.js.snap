// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Account component renders the Account form 1`] = `
<section
  className="AccountPage"
>
  <h2
    className="formTitle"
  >
    Welcome 
    !
  </h2>
  <div
    className="otherButtonBox"
  >
    <NavLink
      className="addRestButton"
      to="/restaurants/hours/add"
    >
      + Add Restaurant
    </NavLink>
  </div>
  <div
    className="noRestaurants"
  >
    You have not added any restaurants at this time...
  </div>
  <br />
  <div
    className="otherButtonBox"
  >
    <NavLink
      className="ownerButton removeButton"
      to="/deleteUser"
    >
      Remove Account
    </NavLink>
  </div>
  <br />
  <br />
</section>
`;
